### ISSUE TITLE
Odd, broken buffering of internal messages...

### ISSUE THREAD

**Issue by fperez** [ipython/ipython#9361]
Consider the following session (this is on master, right now at `51fa9ee`):

```
In [1]: alias_magic foo

In [2]: 1+2
Out[2]: 3

In [3]: exit
UsageError: the following arguments are required: targetdenali[~]>
```

My call to `alias_magic` was missing one argument, but the error message got completely swallowed (even though it's a print statement inside the `alias_magic` code). It only makes it to stdout at shutdown.  I tried with longer sessions, creating more errors, and they just keep piling up internally until the exit.  Very odd.

Ping @takluyver, I suspect this is also related to the recent ptoolkit work.

**Comment by takluyver** [comment]
Can reproduce, will investigate, but this doesn't look like something that would result from the prompt_toolkit changes.

**Comment by takluyver** [comment]
Should be fixed by #9365
